src/promptflow/promptflow/_sdk/_utils.py

```diff
@@ -624,6 +624,31 @@ def _generate_tool_meta(
     return res
 
 
+def _retrieve_tool_func_result(func_call_scenario: str, function_config: Dict):
+    """Retrieve tool func result according to func_call_scenario.
+
+    :param func_call_scenario: function call scenario
+    :param function_config: function config in tool meta. Should contain'func_path' and 'func_kwargs'.
+    :return: func call result according to func_call_scenario.
+    """
+    func_path = function_config.get("func_path", "")
+    func_kwargs = function_config.get("func_kwargs", {})
+
```

Comment:
# May call azure control plane api in the custom function to list Azure resources.
    # which may need Azure workspace triple.
    # TODO: move this method to a common place.

Reply:
updated.